### PR TITLE
 rustdoc: hide macro export statements from docs

### PR DESCRIPTION
**Pull request by QuietMisdreavus**
As mentioned in https://github.com/rust-lang/rust/issues/50647, rustdoc now prints both the import statement and the macro itself when re-exporting macros. This is a stopgap solution to clean up the std docs and get something small backported into beta.

What this does: When rustdoc finds an export statement for a macro, instead of printing the export and bailing, now it will instead hide the export and bail. Until we can solve https://github.com/rust-lang/rust/issues/34843 or have a better way to find the attributes on an export statement when inlining macros, this will at least match the current behavior and clean up the re-export statements from the docs.

### CODE REVIEW

**Comment by QuietMisdreavus** [comment]
Curses, rust-highfive hasn't reloaded the teams list.

r? @GuillaumeGomez

**Comment by GuillaumeGomez** [comment]
That's already a first fix so let's get it in. Thanks!

File: src/test/rustdoc/pub-use-extern-macros.rs [hunk 1]
```diff
@@ -15,6 +15,7 @@
 extern crate macros;
 
 // @has pub_use_extern_macros/macro.bar.html
+// @!has pub_use_extern_macros/index.html 'pub use macros::bar;'
```
Comment (ollie27) [inline src/test/rustdoc/pub-use-extern-macros.rs]: This needs to be something like `// @!has pub_use_extern_macros/index.html '//code' 'pub use macros::bar;'` to actually test this. The other `@!has` tests in this file need to be fixed in the same way.

Comment (QuietMisdreavus) [reply src/test/rustdoc/pub-use-extern-macros.rs]: Oh, that's a good point! It looks like the 2-operator `@has` command doesn't do any HTML parsing. I didn't realize that. `>_>`

**Comment by QuietMisdreavus** [comment]
@bors delegate=ollie27

I'm not sure whether bors has been update with the new r+ permissions, so here's to make sure.

**Comment by QuietMisdreavus** [comment]
Nominating for beta so we can clean up the "Re-exports" section from the std docs.

**Comment by QuietMisdreavus** [comment]
...I suppose to that end, i should ping @rust-lang/rustdoc to make sure everyone agrees on it.

**Comment by pietroalbini** [comment]
Ping @rust-lang/rustdoc! Can someone check this PR and approve it for beta backport?

**Comment by QuietMisdreavus** [comment]
Paging @GuillaumeGomez @ollie27 @steveklabnik! Pending beta backport decision awaits your input!

**Comment by steveklabnik** [comment]
Let's do it.

**Pull request merged**